test/syncenginetestutils.h

```diff
@@ -656,7 +658,7 @@ class FakeFolder
         return rootTemplate;
     }
 
-    FileInfo currentRemoteState() { return _fakeQnam->currentRemoteState(); }
+    FileInfo &currentRemoteState() { return _fakeQnam->currentRemoteState(); }
```

Comment:
Or actually simpler, change the type of remoteModifier to `FileInfo &`, and use that one instead.